src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java

```diff
@@ -43,9 +44,17 @@ public QueryResponseBean(QueryScreen queryScreen, SessionWrapper session) {
             String currentAnswer = currentAnswers.get(key);
 
             // Map the current Answer to the itemset index of the answer
-            Pair<String[], Integer> choicesAndAnswerIndex = queryPromptItem.getItemsetChoicesWithAnswerIndex(currentAnswer);
-            if (queryPromptItem.isSelectOne()) {
-                currentAnswer = choicesAndAnswerIndex.second == -1 ? null : String.valueOf(choicesAndAnswerIndex.second);
+            String[] choiceLabels = null;
+            if (queryPromptItem.isSelect()) {
+                String[] selectedChoices = RemoteQuerySessionManager.extractSelectChoices(currentAnswer);
+                String answerWithChoiceIndices = "";
+                for (int i = 0; i < selectedChoices.length; i++) {
+                    int choiceIndex = ItemSetUtils.getIndexOf(queryPromptItem.getItemsetBinding(), selectedChoices[i]);
+                    answerWithChoiceIndices += choiceIndex == -1 ? "" : String.valueOf(choiceIndex);
+                    answerWithChoiceIndices += i == selectedChoices.length -1 ? "" : RemoteQuerySessionManager.MULTI_SELECT_DELIMITER;
```

Comment:
might be cleaner to put the indices in an array or list and then use `String.join` with the delimiter to get the final result.

Reply:
thanks, made the change